sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -426,3 +426,68 @@ case class MinOf(left: Expression, right: Expression) extends BinaryArithmetic {
 
   override def toString: String = s"MinOf($left, $right)"
 }
+
+case class Pmod(left: Expression, right: Expression) extends BinaryArithmetic {
+
+  override def symbol: String = "pmod"
+
+  override def toString: String = s"pmod($left, $right)"
+
+  protected def checkTypesInternal(t: DataType) =
+    TypeUtils.checkForNumericExpr(t, "pmod")
+
+  override lazy val resolved =
+    childrenResolved && checkInputDataTypes().isSuccess && !DecimalType.isFixed(dataType)
+
+  protected override def evalInternal(left: Any, right: Any) =
+    dataType match {
+      case IntegerType => pmod(left.asInstanceOf[Integer], right.asInstanceOf[Integer])
+      case LongType => pmod(left.asInstanceOf[Long], right.asInstanceOf[Long])
+      case ShortType => pmod(left.asInstanceOf[Short], right.asInstanceOf[Short])
+      case ByteType => pmod(left.asInstanceOf[Byte], right.asInstanceOf[Byte])
+      case FloatType => pmod(left.asInstanceOf[Float], right.asInstanceOf[Float])
+      case DoubleType => pmod(left.asInstanceOf[Double], right.asInstanceOf[Double])
+      case _: DecimalType => pmod(left.asInstanceOf[Decimal], right.asInstanceOf[Decimal])
```

Comment:
Good idea , but seems like the clac result would be a little bit different with the expected result. `pmod(7.2, 4.1), actual: 3.099999999999999, expected: 3.1000000000000005` . I'm keen on keeping it this way if not obvious downside.

``` scala
  private lazy val integral = dataType match {
    case i: IntegralType => i.integral.asInstanceOf[Integral[Any]]
    case i: FractionalType => i.asIntegral.asInstanceOf[Integral[Any]]
    case i: DecimalType => i.asIntegral.asInstanceOf[Integral[Any]]
  }
  protected override def evalInternal(a: Any, n: Any) = 
    integral.rem(integral.plus(integral.rem(a, n), n), n)
```